src/HotChocolate/Core/src/Types/Types/Scalars/DateTimeType.cs

```diff
@@ -192,7 +192,7 @@ private static bool TryDeserializeFromString(
             && DateTime.TryParse(
                 serialized,
                 CultureInfo.InvariantCulture,
-                DateTimeStyles.AssumeUniversal,
+                DateTimeStyles.AdjustToUniversal | DateTimeStyles.AssumeUniversal,
                 out var zuluTime))
```

Comment:
I think it would be fine to replace this clause with `DateTimeOffset.TryParse`...